repo/open.go

```diff
@@ -247,23 +247,26 @@ func openWithConfig(ctx context.Context, st blob.Storage, cliOpts ClientOptions,
 		return nil, errors.Wrap(ferr, "unable to create format manager")
 	}
 
-	if _, err := retry.WithExponentialBackoffMaxRetries(ctx, -1, "wait for upgrade", func() (interface{}, error) {
-		uli, err := fmgr.UpgradeLockIntent()
+	if !cliOpts.ReadOnly {
```

Comment:
This makes sense. Do we also need to prevent the background lock monitor from interrupting reads when a lock is in place ?